OrbitBase/Platform/Windows/Profiling.cpp

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// clang-format off
+#include <OrbitBase/Platform/Windows/Profiling.h>
+// clang-format on
+
+#include <processthreadsapi.h>
+
+#include <OrbitBase/Logging.h>
+#include <string>
+
+template <typename T>
```

Comment:
Since it puzzled me for a second, can we name `T` something more descriptive, like `FunctionT` or similar?